### ISSUE TITLE
Regression is ruleset testing

### ISSUE THREAD

**Issue by J0WI** [EFForg/https-everywhere#6543]
There are multiple regressions in our testing process:
This commit contains duped targets, but Travis passed: https://travis-ci.org/EFForg/https-everywhere/builds/154001093
Here were also two identical hosts merged, but I'm not sure if Travis was able to detect that, beacause both lived in a separate PR: #6489
This commit contains a test url that doesn't match any target or exclusion: https://travis-ci.org/EFForg/https-everywhere/builds/154047712
On #6426 we have the problem again, that a slow host breaks the fetch test. Referring to the log, the fetch was tried only once.

I think we had all these tests before, so this is likely a regression of the last few changes that were made. I already had a look at the recent changes, but the test code itself didn't change much. Maybe some tests are now running in a wrong context or not at all?
cc @Hainish @cooperq

**Comment by Hainish** [comment]
I see this and will investigate today

**Comment by Hainish** [comment]
Separating these into separate issues

**Comment by Hainish** [comment]
> On #6426 we have the problem again, that a slow host breaks the fetch test. Referring to the log, the fetch was tried only once.

The fetch is run twice, it just isn't printed twice.  It only prints when it is finished comparing, even if it takes two tries.

You can verify this by applying the diff in the attached patch and then running the fetch test.
[diff.patch.txt](https://github.com/EFForg/https-everywhere/files/431517/diff.patch.txt)

**Comment by Hainish** [comment]
> Here were also two identical hosts merged, but I'm not sure if Travis was able to detect that, beacause both lived in a separate PR: #6489

I don't think we can do anything about two PRs conflicting with each other, but neither conflicting with `master`.
